### ISSUE TITLE
Schema with empty name for anonymous types

### ISSUE THREAD

**Issue by dolmen** [invopop/jsonschema#20]
```go
schema := jsonschema.Reflect((*struct {
	T []struct {
		Key   string      `json:"key"`
		Value interface{} `json:"value"`
	} `json:"table"`
})(nil))
```
produces (see [on the Go playground](https://go.dev/play/p/ZrAO-D35K57)):
```json
{
  "$schema": "http://json-schema.org/draft/2020-12/schema",
  "$ref": "#/$defs/",
  "$defs": {
    "": {
      "properties": {
        "table": {
          "$ref": "#/$defs/"
        }
      },
      "additionalProperties": false,
      "type": "object",
      "required": [
        "table"
      ]
    }
  }
}
```

The schema is recursive while the original Go type isn't. It looks like there is a problem in handling of anonymous types. Even more, when there is multiple anonymous types the result is completely corrupted.

For anonymous types I expect an inline schema, not a schema defined via a reference under `#/$defs`.

Expected schema:
```json
{
  "$schema": "http://json-schema.org/draft/2020-12/schema",
  "additionalProperties": false,
  "properties": {
    "table": {
      "items": {
        "additionalProperties": false,
        "properties": {
          "name": {
            "type": "string"
          },
          "value": {
          }
        },
        "required": [
          "name",
          "value"
        ],
        "type": "object"
      },
      "type": "array",
    }
  },
  "required": [
    "table"
  ],
  "type": "object"
}
```

**Comment by samlown** [comment]
This one looks fixed to me with `v0.5.0`, please confirm!

**Comment by samlown** [comment]
Assuming this one is fixed! Let us know if not. cheers.
